src/main/java/com/aws/greengrass/clientdevices/auth/certificate/infra/BackgroundCertificateRefresh.java

```diff
@@ -122,16 +127,44 @@ public boolean isRunning() {
      * Runs verifyIotCertificate useCase for all the registered client certificate PEMs.
      */
     @Override
-    public void run() {
+    public synchronized void run() {
         if (isNetworkDown()) {
             logger.debug("Network is down - not refreshing certificates");
             return;
         }
 
+        if (!canRun()) {
+            logger.info("Certificates were refreshed recently - retrying later");
+            return;
+        }
```

Comment:
Customer networks can be quite chatty - this isn't appropriate for INFO level. I'd actually suggest just removing the log entirely and only log when we do refresh.